packages/expo-cli/src/commands/eas-build/init/action.ts

```diff
@@ -0,0 +1,109 @@
+import chalk from 'chalk';
+import figures from 'figures';
+import fs from 'fs-extra';
+import ora from 'ora';
+import path from 'path';
+
+import { EasConfig, EasJsonReader } from '../../../easJson';
+import log from '../../../log';
+import AndroidBuilder from '../build/builders/AndroidBuilder';
+import iOSBuilder from '../build/builders/iOSBuilder';
+import { BuildCommandPlatform } from '../types';
+import createBuilderContextAsync from '../utils/createBuilderContextAsync';
+import {
+  DirtyGitTreeError,
+  ensureGitStatusIsCleanAsync,
+  reviewAndCommitChangesAsync,
+} from '../utils/git';
+
+interface BuildOptions {
+  platform: BuildCommandPlatform;
+  skipCredentialsCheck?: boolean; // TODO: noop for now
+  profile: string;
+  parent?: {
+    nonInteractive: boolean;
+  };
+}
+
+async function initAction(projectDir: string, options: BuildOptions): Promise<void> {
```

Comment:
I'm not 100% sure what is commanderjs behavior where, but from what I remember if you run
`expo eas:init --profile` then this action would be triggered with options set to  `{ profile: true }`

Make sure if I'm correct and handle that case if necessary